include/wil/networking.h

```diff
@@ -0,0 +1,1487 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+//! @file
+//! Helpers for using BSD sockets and Winsock functions and structures.
+//! Does not require the use of the STL or C++ exceptions (see _nothrow functions)
+#ifndef __WIL_NETWORKING_INCLUDED
+#define __WIL_NETWORKING_INCLUDED
+
+#ifdef _KERNEL_MODE
+#error This header is not supported in kernel-mode.
+#endif
+
+//! define WIN32_LEAN_AND_MEAN at the project level to avoid windows.h including older winsock 1.1 headers from winsock.h
+//! as including both the Winsock 1.1 header 'winsock.h' and the Winsock 2 header 'winsock2.h' will create compiler errors
+//! alternatively, including winsock2.h before windows.h to prevent inclusion of the Winsock 1.1 winsock.h header from within windows.h
+//! note: winsock2.h will include windows.h if not already included
+//!
+//! define _SECURE_SOCKET_TYPES_DEFINED_ at the project level to have access to SocketSecurity* functions in ws2tcpip.h
+//!
+//! define INCL_WINSOCK_API_TYPEDEFS at the project level to make function typedefs available across various networking headers
+//! note, winsock2.h defaults is to not include function typedefs - but these can be necessary when supporting multiple OS versions
+//!
+//! Link libs for functions referenced in this file: ws2_32.lib, ntdll.lib, and Fwpuclnt.lib (for secure socket functions)
+
+#if !defined(_WINSOCK2API_) && defined(_WINSOCKAPI_)
+#error The Winsock 1.1 winsock.h header was included before the Winsock 2 winsock2.h header - this will cause compilation errors - define WIN32_LEAN_AND_MEAN to avoid winsock.h included by windows.h

```

Comment:
The comment above says the problem is avoided by `#define`'ing `WIN32_LEAN_AND_MEAN` -OR- `#include`'ing `winsock.h` before `Windows.h` but the error message only says the former. The error message should include both options